backend/src/db/providers/courseProvider.ts

```diff
@@ -41,13 +41,19 @@ const getCourseProvider = async (courseId: string) => {
 };
 
 const applyCourseProvider = async (courseId: string, userId: string) => {
-  const applyedCourse = await ClientCourseModel.create({
-    course: courseId,
-    status: 'approved',
-    currentStage: 1,
-  });
-  await UserModel.updateOne({ _id: userId }, { $push: { courses: applyedCourse.course } });
-  return applyedCourse;
+  const clientCourse = await ClientCourseModel.findOne({ course: courseId });

```

Comment:
what you are looking for in this collection? 
courseId is from another collection 'courses', you won't find it